### PR TITLE
Fixed: Escape regex in ParseAlbumWithSearchCriteria

### LINKED ISSUE #231: Alternative parsing should escape strings before building regex.
https://sentry.io/lidarr/lidarr/issues/425604862/

```
ArgumentException: parsing "\b(?<artist>Ed Sheeran)\b.*\b(?<album>+)\b" - Quantifier {x,y} following nothing.
  File "\NzbDrone.Core\Parser\Parser.cs", line 349, in ParseAlbumTitleWithSearchCriteria
    NzbDrone.Core.Parser.Model.ParsedAlbumInfo ParseAlbumTitleWithSearchCriteria(System.String, NzbDrone.Core.Music.Artist, System.Collections.Generic.List`1[NzbDrone.Core.Music.Album])

An error has occurred while trying to parse Ed Sheeran I See Fire[Mimp3.eu].mp3
```

### PR DESCRIPTION
**Pull request by danielunderwood**
#### Database Migration
NO

#### Description
Adds regex escape to `Parser.ParseAlbumWithSearchCriteria` to fix #231.

#### Issues Fixed or Closed by this PR

* #231

### CODE REVIEW

File: src/NzbDrone.Core/Parser/Parser.cs [hunk 1]
```diff
@@ -338,7 +338,7 @@ public static ParsedAlbumInfo ParseAlbumTitleWithSearchCriteria(string title, Ar
 
                 simpleTitle = CleanTorrentSuffixRegex.Replace(simpleTitle, string.Empty);
 
-                var releaseRegex = new Regex(@"\b(?<artist>" + artist.Name + @")\b.*\b(?<album>"+ string.Join("|",album.Select(s=>s.Title).ToList()) + @")\b",
+                var releaseRegex = new Regex(@"\b(?<artist>" + Regex.Escape(artist.Name) + @")\b.*\b(?<album>"+ string.Join("|",album.Select(s=>Regex.Escape(s.Title)).ToList()) + @")\b",
```
Comment (Qstick) [inline src/NzbDrone.Core/Parser/Parser.cs]: This works to fix problem in most cases, however if artist or album is just a symbol then the word boundary will not work. Can we change this to build the regex string in two parts? Check artist for symbol only then build it, check album for symbol then build it. Then combine strings and build the full regex to check releases.

File: src/NzbDrone.Core/Parser/Parser.cs [hunk 2]
```diff
@@ -338,8 +339,36 @@ public static ParsedAlbumInfo ParseAlbumTitleWithSearchCriteria(string title, Ar
...
+                {
+                    albums = string.Join(" ", album.Select(s => s.Title).ToList());
+                } else
+                {
+                    albums = string.Join("|", album.Select(s => s.Title).ToList());
+                }
+                var escapedArtist = Regex.Escape(artist.Name);
+                var escapedAlbums = Regex.Escape(albums);
+                StringBuilder albumBuilder = new StringBuilder();
+                foreach(char c in albums)
```
Comment (Qstick) [inline src/NzbDrone.Core/Parser/Parser.cs]: What’s this for? Doesn’t the regex.escape(albums) above handle this?

**Comment by Qstick** [comment]
5 additional tests are failing

File: src/NzbDrone.Core/Parser/Parser.cs (hunk 2 above)
Comment (majora2007) [reply src/NzbDrone.Core/Parser/Parser.cs]: No, not on the case of /// \\\. Hence I employed a custom solution. If you are fine with just that case failing, I can remove.

**Comment by majora2007** [comment]
Fixed up the tests. Please re-review @Qstick

**Pull request merged**